Automattic/wp-calypso
# Move parts of lib/url to calypso-url package

#### Changes proposed in this Pull Request

As part of my goal of moving large amounts of checkout into a package outside of calypso so that it can have type safety and be used in wp-admin, I've been working to put checkout's major dependencies into packages first. Probably the biggest of those is the combination of `lib/products-values` and `lib/plans`, but those directories have a dependency on `lib/url`, which has very few dependencies of its own. This PR therefore begins to move `lib/url` into the package `@automattic/calypso-url`. To start with, it only moves the code required by `lib/plans`, however, which is the following functions:

- `format`
- `getUrlParts`
- `getUrlFromParts`
- `determineUrlType`
- and the enum `URL_TYPE`

I'm focusing on these functions (which do seem to be the core of that directory anyway, according to its README) because moving the entirety of `lib/url` would require dealing with quite a few other dependencies, which are really out of the scope of this project (but could be migrated later).

#### Testing instructions

This does not change any of the functions that are moved, and it attempts to update all imports of those functions. If there is any import that was missed, the calypso build should fail. Therefore automated tests should be sufficient to prove that these changes are safe, but I'd appreciate any other opinions.
